data_structures/src/serializers.rs

```diff
@@ -172,6 +192,88 @@ impl TryFrom<Vec<u8>> for Message {
                     }
                 })
                 .ok_or(""),
+            FlatBuffersCommand::Block => message
+                .command_as_block()
+                .map(|block| {
+                    // Get Header
+                    let header_ftb = block.header();
+                    let version = header_ftb.version();
+                    // Get CheckpointBeacon
+                    let hash: Hash = match header_ftb.beacon().hash_prev_block().type_() {
```

Comment:
This can be refactored into a `create_hash` function as well as `create_address`